packages/core/src/routes/session/utils.ts

```diff
@@ -152,6 +152,13 @@ export const getContinueSignInResult = async (
   return rest;
 };
 
+export const isUserPasswordSet = ({
+  passwordEncrypted,
+  identities,
+}: Pick<User, 'passwordEncrypted' | 'identities'>): boolean => {
+  return Boolean(passwordEncrypted) || Object.keys(identities).length > 0;
```

Comment:
nit: do we need to specify specific error for social identities exist error?

Reply:
This error is not expected to happen in a normal flow, so keep it for now.